crates/subspace-farmer/src/plot.rs

```diff
@@ -420,42 +434,63 @@ impl WeakPlot {
 struct IndexHashToOffsetDB {
     inner: DB,
     address: PublicKey,
-    max_distance_key: Option<PieceDistance>,
+    max_distance_cache: BTreeSet<BidirectionalDistanceSorted<PieceDistance>>,
```

Comment:
`max_distance_cache` could benefit from its own abstraction too.

Reply:
Decided to leave it as it is, because implementing autonomous update requires passing iterator or closure, which would initialize iterator lazily. That would make api cumbersome.